src/agent.ts

```diff
@@ -86,15 +86,20 @@ export class Agent {
    */
   async run(
     input: string,
-    { model, network }: Agent.RunOptions | undefined = {},
+    {
+      model,
+      network,
+      maxIter = this.tools.size + 1,
```

Comment:
Are we expecting _every_ tool to be called _exactly_ once here?  This seems really hard to tell up front 🤔